realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -181,6 +198,10 @@ public static synchronized SyncSession getSession(SyncConfiguration syncConfigur
         if (session == null) {
             session = new SyncSession(syncConfiguration);
             sessions.put(syncConfiguration.getPath(), session);
+            if (sessions.size() == 1) {
+                RealmLog.debug("first session created add network listener");
+                NetworkStateReceiver.addListener(networkListener);
```

Comment:
`NetworkStateReceiver.addListener()` is not `synchronized` method, but `NetworkStateReceiver.removeListener()` is . Is it correct?